### PR TITLE
Add stacktraces to EventHandler error logging

### PR DESCRIPTION
**Pull request by davydog187**
### Issue
#303 

### Description
This PR logs the stacktrace when a user-written EventHandler raises an error. 

Note that I hacked in a test scenario, there might be a better way of doing that. I'm open to suggestions.

Screen Shot 2020-02-16 at 9 35 45 AM

### CODE REVIEW

File: lib/commanded/event/handler.ex [hunk 1]
```diff
@@ -626,6 +626,15 @@ defmodule Commanded.Event.Handler do
         end)
 
         handle_event_error(error, event, state, context)
+
+      {:error, reason, stacktrace} ->
+        Logger.error(fn ->
+          describe(state) <>
+            " failed to handle event #{inspect(event)} due to: #{inspect(reason)}"
```
Comment (davydog187) [inline lib/commanded/event/handler.ex]: In the future, I would suggest we move to the Exception formatter to log these. I can open another issue if we think that's a good idea.

File: lib/commanded/event/handler.ex [hunk 2]
```diff
@@ -626,6 +626,15 @@ defmodule Commanded.Event.Handler do
...
+        end)
+        Logger.error(Exception.format(:error, reason, stacktrace))
```
Comment (davydog187) [inline lib/commanded/event/handler.ex]: I should use the function style logs

File: lib/commanded/event/handler.ex (hunk 1 above)
Comment (davydog187) [reply lib/commanded/event/handler.ex]: I added the `pretty` option to inspect the error messages. It makes the log much easier to read.

Screen Shot 2020-02-17 at 7 26 21 AM

**Comment by slashdotdash** [comment]
Thanks Dave.

**Pull request merged**